lambdaclass/mirra_backend
# [CONFIGURATOR] Versioning + UI

## Motivation

This is part of #1002 
Closes #1029
Closes #1030

## Summary of changes

- [Add versioning feature (1st iteration)](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/3f79fd5ef89cb91c3cd6d2b559b763d7c7203e2a) 
- [Add 'back to home' redirect button](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/c5a1f73c33de87cc30b22c39a212da27141ed617)
- [Fix all character views to work with new versions' behavior](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/90f5efe5d5e82039ec85ae690b5e8c7ad7f036c8)
- [Fix all game_configuration views to work with new versions' behavior](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/013c13a43b9c881bb01fd32e7f45c9bb470f19e7)
- [Fix error cases in character controller](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/c44746289007a5af9ffb56c8e7b481709c2aad13)
- [Fix all map_configuration views to work with new versions' behavior](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/fca0561da438c29b1f3db385cda025aaa5f8efec)
- [Fix all skills views to work with new versions' behavior](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/e4af62b43c236541becd0a135d19e28e2f0bcda5)
- [Fix all consumable_items views to work with new versions' behavior](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/75e3a1de037a8e75653527849dbc955e17b2d689)
- [Fix seeds for both autobattler and arena apps, improve UI styles and …](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/d3ddfb69afa2b082478d17d12a55f68e6b46de1f) 
- [Add gh issue link to TODOs](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/0fc788e780233a9e8cffdd917351e26b59a1c9f0)
- [Refactor function to list curse skills grouped by type](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/b1fab3c862f407bcfee347a18df9e3eb66178b00)
- [Current version now takes all consumable items, also handle items wit…](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/58525aa6e18c950264d63905c7607e2dac4ae6fc) 
- [Filter active items when encoding the configuration](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/c61e535e86fbd96647471b8c811a34466afdd3d3)
- [Rename :version_id path param due to override and fix warnings in code](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/5593d9a5ea42dcd3a2eb25d8bfb6de9ac5751ae2)
- [Add missing param in map_confguration_form](https://github.com/lambdaclass/mirra_backend/pull/1013/commits/2f707663662b1177fa3a7e284d3c7a079a74aa32)

## How to test it?

To create a new version:
- New version button
- Go all way down and submit the form.
- In versions index, you can see all the versions + the one you just created.
- You can see everything in it, edit it, etc.

What to check:
- The game will use the current one, if you mark as current a new one, everything should work as expected.
- Everything should be properly associated, without losing any record.

Known issues and next steps:
https://github.com/lambdaclass/mirra_backend/issues/1028

## Checklist
